bob-apps/bin/bobd.rs

```diff
@@ -120,6 +148,115 @@ async fn main() {
+        sub_matches.value_of("data").unwrap_or("data").to_string(),
+        addresses).unwrap();
+    let http_api_port = sub_matches.value_of("restapi-port").and_then(|v|  Some(v.parse().unwrap()));
+    let node = cluster.get_testmode_node(this_node.unwrap(), http_api_port).unwrap();
+
+    init_testmode_logger(log::LevelFilter::Info);
+
+    check_folders(&node, true);
+
+    println!("Bob has started");
+    let n = &cluster.nodes()[0];
+    println!("Data directory: {}", n.disks()[0].path());
+    println!("gRPC API available at: {}", n.address());
+    let a = node.http_api_address();
+    let p = node.http_api_port();
+    println!("REST API available at: http://{}:{}", a, p);
+    println!("REST API Put and Get available at: http://{}:{}/data", a, p);
+
+    Ok((cluster, node))
+}
+
+fn init_testmode_logger(loglevel: log::LevelFilter) {
+    let stdout = ConsoleAppender::builder()
+        .encoder(Box::new(PatternEncoder::new( "{d(%Y-%m-%d %H:%M:%S):<20} {M:>20.30}:{L:>3} {h({l})}    {m}\n")))
+        .build();
+
+    let _20mb = 20000000;
+    let trigger = SizeTrigger::new(_20mb);
+    let roller = FixedWindowRoller::builder()
+        .build( "./log/archive/info.{}.log", 10).unwrap();
+    let policy = CompoundPolicy::new(Box::new(trigger), Box::new(roller));
+    let requests_error = RollingFileAppender::builder()
+        .encoder(Box::new(PatternEncoder::new("{d(%Y-%m-%d %H:%M:%S):<20} {M:>20.30}:{L:>3} {l} {m}{n}")))
+        .build( "./log/error.log", Box::new(policy)).unwrap();
+
+    let config = Config::builder()
+        .appender(Appender::builder().build("stdout", Box::new(stdout)))
+        .appender(Appender::builder().build("requests_error", Box::new(requests_error)))
+        .logger(Logger::builder()
+                .appender("stdout")
+                .appender("requests_error")
+                .additive(false)
+                .build("bob", loglevel))
+        .logger(Logger::builder()
+                .appender("stdout")
+                .appender("requests_error")
+                .additive(false)
+                .build("bob_backend", loglevel))
+        .logger(Logger::builder()
+                .appender("stdout")
+                .appender("requests_error")
+                .additive(false)
+                .build("bob_common", loglevel))
+        .logger(Logger::builder()
+                .appender("stdout")
+                .appender("requests_error")
+                .additive(false)
+                .build("bob_apps", loglevel))
+        .logger(Logger::builder()
+                .appender("stdout")
+                .appender("requests_error")
+                .additive(false)
+                .build("bob_grpc", loglevel))
+        .logger(Logger::builder()
+                .appender("stdout")
+                .appender("requests_error")
+                .additive(false)
+                .build("pearl", loglevel))
+        .build(Root::builder().appender("stdout").build(loglevel))
```

Comment:
Maybe it would be enough to create only a Root logger? Can you check if logs will be written if only the root logger is created?

Reply:
Yes: [0cf1f5f](https://github.com/qoollo/bob/pull/588/commits/0cf1f5f52ef2601db946f55f99307fb75336e6c7)